### PR TITLE
fix(core): avoided creating standalone exit spans when using the SDK

### PR DESCRIPTION
**Pull request by kirrg001**
refs https://jsw.ibm.com/browse/INSTA-10132

### CODE REVIEW

**Comment by kirrg001** [comment]
This is not an easy fix. I hope it won't break other SDK use cases 🤞

File: packages/core/src/tracing/clsHooked/unset.js [hunk 1]
```diff
@@ -55,8 +55,12 @@ function storeReducedSpan(context, key, span) {
...
   // around forever. By keeping the reduced span we can at least see (in a heap dump) for which type of spans the
   // destroy call is missing, aiding in troubleshooting the Node.js runtime bug.
-
-  if (key === currentSpanKey && span != null) {
+  //
+  // Exception:
+  // SDK spans are different. Customers use "startEntrySpan" or "startExitSpan" manually inside their code base.
+  // We cannot remember the reduced spans because it can mislead to a wrong functionality such as
+  // tracing exit spans without entry spans - see sdkApp1.js.
+  if (key === currentSpanKey && span != null && span.n !== 'sdk') {
```
Comment (kirrg001) [inline packages/core/src/tracing/clsHooked/unset.js:63]: @aryamohanan Its important that you understand the purpose of the unset function & remembering reduced spans to judge the change.

File: packages/collector/test/tracing/protocols/http/client/test.js [hunk 2]
```diff
@@ -39,6 +39,59 @@ mochaSuiteFn('tracing/http client', function () {
...
+      sdkControls = new ProcessControls({
+        appPath: path.join(__dirname, 'sdkApp1'),
+        useGlobalAgent: true,
+        env: {}
+      });
+
+      await sdkControls.start(null, null, true);
+    });
+
+    it('should not trace example.com exit span without entry span', async () => {
```
[the remark this answers is not in the document]
Comment (kirrg001) [reply packages/collector/test/tracing/protocols/http/client/test.js:56]: Thanks. Will take a look

File: packages/core/src/tracing/clsHooked/unset.js (hunk 1 above)
Comment (aryamohanan) [reply packages/core/src/tracing/clsHooked/unset.js:63]: I need some more time for reviewing this PR.

File: packages/collector/test/tracing/protocols/http/client/test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirrg001) [reply packages/collector/test/tracing/protocols/http/client/test.js:56]: Working now! Await was missing

[the remark this answers is not in the document]
Comment (kirrg001) [reply packages/collector/test/tracing/protocols/http/client/test.js:56]: Thanks for catching @aryamohanan <3

**Comment by aryamohanan** [comment]
LGTM

**Review by aryamohanan — APPROVED** [review]

**Pull request merged**